plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -209,7 +210,7 @@ public Transaction newCreateTableTransaction(
     }
 
     @Override
-    public void registerTable(ConnectorSession session, SchemaTableName tableName, String tableLocation, String metadataLocation)
+    public void registerTable(ConnectorSession session, SchemaTableName tableName, String tableLocation, TableMetadata tableMetadata, String metadataLocation)
```

Comment:
`TableMetadata` contains `metadataFileLocation` method. Why do we want to pass two arguments?

Reply:
Actually the tableLocation is also obsolete (and most catalogs don't use it). Removed in additional commit, but also added verification.